src/makeInfinityAdder.js

```diff
@@ -1,12 +1,30 @@
 'use strict';
 
 /***
- *
  * @return {function}
  */
-
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+  let result = 0;
+
+  const adder = (a) => {
+    if (typeof a !== 'number') {
+      if (!result) {
+        result = sum;
+      } else {
+        sum = 0;
+        result = 0;
```

Comment:
This logic will not work as expected. If 'result' is falsy (which will be the case initially), it will always set 'result' to 'sum' and never reset 'sum' to 0. You need to adjust this logic to reset 'sum' when no arguments are provided.